pkg/cmd/repo/autolink/list/list_test.go

```diff
@@ -0,0 +1,247 @@
+package list
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/internal/browser"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/google/shlex"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewCmdList(t *testing.T) {
+	tests := []struct {
+		name         string
+		input        string
+		output       listOptions
+		wantErr      bool
+		wantExporter bool
+		errMsg       string
+	}{
+		{
+			name:   "no argument",
+			input:  "",
+			output: listOptions{},
+		},
+		{
+			name:   "web flag",
+			input:  "--web",
+			output: listOptions{WebMode: true},
+		},
+		{
+			name:         "json flag",
+			input:        "--json id",
+			output:       listOptions{},
+			wantExporter: true,
+		},
+		{
+			name:    "invalid json flag",
+			input:   "--json invalid",
+			wantErr: true,
+			errMsg: heredoc.Doc(`
+				Unknown JSON field: "invalid"
+				Available fields:
+				  id
+				  isAlphanumeric
+				  keyPrefix
+				  urlTemplate`),
+		},
+	}
+
```

Comment:
**Suggestion**

Instead of this, I'd like to see a test like https://github.com/cli/cli/blob/48bc3cababce65bda8e185f28a467404daf7a76c/pkg/cmd/issue/view/view_test.go#L25-L48